### ISSUE TITLE
[Scheduler] Recurrent tasks are not displayed properly on the day of the DST switch

### ISSUE THREAD

**Issue by Merrique** [syncfusion/ej2-vue-ui-components#100]
Hello, 

Recently I started to investigate the way the recurrent tasks are being displayed in the scheduler on the day of the DST change and there is one situation which I believe is working incorrectly. 

For the timezone "Europe/Warsaw" I have created two recurrent tasks starting on 22nd Oct 2023 and ending on 22nd Nov 2023. One tasks starts at 12am and finishes at 3am and the other one starts at 12am and finishes at 2am. In Warsaw on the 29th of Oct there is a change from Summer to Winter time (3am -> 2am) and this is a day I found unexpected behaviour. I will refer to two tasks created on this specific day as  "Task 12-2am" and  "Task 12-3am"

In the database the Task 12-3am has following details:
```
{
startDatetime: "2023-10-28T22:00:00Z"
endDatetime: "2023-10-29T01:00:00Z
duration: 180 // value in minutes
}
```
And the Tasks 12-2am is defined as:
```
{
startDatetime: "2023-10-28T22:00:00Z"
endDatetime: "2023-10-29T00:00:00Z
duration: 120 // value in minutes
}
```

While creating `dataSource` for the scheduler I used both the strings in the format given directly from the databse (i.e. values in UTC as well as after converting them to the local timezone using method

```
moment.tz(datetime_from_database, "Europe/Warsaw" ).format()
```
Where `datetime_from_database` where simply the startDatetime or endDatetime of the specific task. 
For both values the result has been the same. 

The results can be seen on image below.
![Screenshot from 2023-06-14 13-58-17](https://github.com/syncfusion/ej2-vue-ui-components/assets/79991096/a9abf6dd-25d5-4373-97e3-ae697c505cdb)

Everything works as expected for the Task 12-2am. However the task 12-3am for some reason is being displayed as if it would have 1h duration.. The expected behaviour would be to display it as identical to Task 12-2am since we are moving 3am -> 2am on the 29th Oct. 

I am attaching a screenshot with identical situation created for Google Calendar. 
![Screenshot from 2023-06-14 14-01-41](https://github.com/syncfusion/ej2-vue-ui-components/assets/79991096/f6b3ef2a-6759-4781-8645-e6eae163fe74)

Thank you in advance for checking this out.

**Comment by maheshtps** [comment]
Hi Merrique,

We would like to inform you that, this is the default behavior of our Scheduler UI design. Please get back to us if you need any further assistance.

Regards,
Mahesh

**Comment by Merrique** [comment]
Hello,

Thank you for your feedback and information that this is an expected behaviour. 

Closing an issue as this is not a bug :)

**Comment by ravikumar-venkatesan-m** [comment]
Hi Merrique,

Please ignore our last update.

Sample: https://codesandbox.io/s/ej2-vue-3-schedule-recurrence-appointment-in-dst-timezone-kqxynq

We have validated your reported problem at our end by preparing a sample based on your shared details. As per your shared details, we suspect that you are facing your reported problem in the time zone shown in the shared snip. But we are unable to reproduce your reported problem in this time zone. Could you please check our shared sample if you still facing the same problem share the below details to proceed further.

- Your system time zone details.
- If you are using the timezone property of the Schedule share the timezone value that you are using.
- Issue reproducing video demo.
- Reproduce your reported problem in our shared sample or
- Schedule-related code snippets with the codes that you are using to generate appointments and bind them to the Schedule or
- Share an issue reproducing a sample.
- Share any additional details which will help us to reproduce the issue at our end.

 
![image](https://github.com/syncfusion/ej2-vue-ui-components/assets/59193715/a74e6d90-e84d-45f4-a7f0-c152621d28da)


Regards,
Ravikumar Venkatesan

**Comment by Merrique** [comment]
Hello @ravikumar-venkatesan-m ,

I have played a little bit with the set up you have prepared. 
And when running the things as they are done by default everything looks good:
![no timezone prop](https://github.com/syncfusion/ej2-vue-ui-components/assets/79991096/01170511-5364-425b-a969-0bed606d3f13)

The situation changes to the problem I have described when I add prop `timezone` to the `ejs-schedule`. 
This very strange to me as the timezone given is the same timezone as the timezone of my computer and it agrees to the timezone you added when doing calculation using moment.js
![added timezone prop](https://github.com/syncfusion/ej2-vue-ui-components/assets/79991096/6b2e18d5-6e8d-4612-89f3-675c50afc7c3)
I do add the timezone in the application I work in as user can change it freely depending on its need.

There is one more thing which is important and it's the day when we start the recurrence. When I changed the start of the repetition to be week earlier, i.e. not on the day of DST than the task on 29th does not longer "feel" that its the day of the switch between summer and winter time. 
This is the image taken without added timezone prop, but you can notice the change to the beginning of repetition pattern. 
![changed repetition start week earlier](https://github.com/syncfusion/ej2-vue-ui-components/assets/79991096/f60e7eee-6341-4871-a300-cf8fd8db652d)

After adding the timezone prop the tasks are identical:
![changed repetition start week earlier with timezone prop](https://github.com/syncfusion/ej2-vue-ui-components/assets/79991096/2cd47b2c-47aa-48ed-95be-86285e847234)

The part related to the start of the recurrence is not important for me as I have access to all the tasks of recurrence as if they were single tasks, thus I never add `RecurrenceRule`

**Comment by Merrique** [comment]
But I guess I can make it work if I will stop using the timezone prop what is very unfortunate in my opinion as it will require changing the datasource every time I change timezone.

**Comment by Merrique** [comment]
Hello again @ravikumar-venkatesan-m,
I have testes the idea I mentioned in the message before and actually it will not work. Since without providing timezone prop even if I recalculate the `startTime` to be in the desired timezone it is later converted to my local timezone

![Screenshot from 2023-07-12 09-43-04](https://github.com/syncfusion/ej2-vue-ui-components/assets/79991096/3998ff42-f292-4571-8c6f-15b24f25b7c2)

**Comment by SwathiRavi14** [comment]
Merrique,

We have validated your reported issue by altering the sample based on the information you provided. However, its working fine as expected. We checked the identical issue in Outlook calendar and discovered that it functions similarly to Schedule. We designed our Schedule component in accordance with Outlook requirements.
 
Sample: https://codesandbox.io/s/ej2-vue-3-schedule-recurrence-appointment-in-dst-timezone-forked-2tfdpf?file=/src/App.vue
 
Outlook:
 
 
![image](https://github.com/syncfusion/ej2-vue-ui-components/assets/102658583/60104d6b-52ad-43a1-9f85-7ed1b3ceaebe)


 
 
Schedule:
 
 
![image](https://github.com/syncfusion/ej2-vue-ui-components/assets/102658583/7e02cabc-eb6a-43c7-b864-b18d716e721a)



![image](https://github.com/syncfusion/ej2-vue-ui-components/assets/102658583/bc7cdcce-a949-4eb7-8f67-75fa12cc3968)

**Comment by Merrique** [comment]
Hello @SwathiRavi14,
The issue is still persistent for me if you set the start of the original task at the day of the DST.

No repetition task scheduled at 29th of October (changed the start date and end date and removed recurrence)
![Screenshot from 2023-07-19 09-55-54](https://github.com/syncfusion/ej2-vue-ui-components/assets/79991096/43858674-12bc-4e62-a551-23341e83de29)

Adding repetition but for the task that starts on the 29th of October. Please notice that the "1h" task (12-3am) is repeated in the future.
![Screenshot from 2023-07-19 09-56-42](https://github.com/syncfusion/ej2-vue-ui-components/assets/79991096/292fbdeb-3b63-4c61-8edd-f82db99591b3)
![Screenshot from 2023-07-19 10-01-09](https://github.com/syncfusion/ej2-vue-ui-components/assets/79991096/cbd0c842-6f19-41ac-a747-13e8956a742b)

This is the source of my issue. In the project I'm working on for various reason all of the tasks are being fetched even if they were generated as the recurrent task. Which means I read it as a single task ad the `RecurrenceRule` is never added to my data source. I guess my description was misleading at the start because of this.

**Comment by SwathiRavi14** [comment]
@Merrique ,

We checked your reported issue in our Scheduler and informed you that the **StartTimeZone** and **EndTimeZone** fields in your appointments data are required. Regarding your observation about tasks on the day of Daylight-Saving Time (DST), we have analyzed the behavior of the Scheduler in comparison to the Outlook calendar. Our investigation shows that both applications handle recurring events in a similar manner, and the repeating occurrences are shown with the same start and end times as the parent recurring event. 

**Sample**: https://codesandbox.io/s/ej2-vue-3-schedule-recurrence-appointment-in-dst-timezone-forked-tk2wl8

**Schedule**:

 
![image](https://github.com/syncfusion/ej2-vue-ui-components/assets/102658583/1fef11c7-579b-4a0b-80d0-c8c09c152c86)



**Outlook**:

 
![image](https://github.com/syncfusion/ej2-vue-ui-components/assets/102658583/95b88aa9-b5e0-4269-9d5d-ab42b945e7a0)

**Comment by Merrique** [comment]
@SwathiRavi14,

The images i have provided were generated after making changes to your codesandbox, where no `StarTimeZone` or `endTimeZone` were provided. The link you attached is here https://codesandbox.io/s/ej2-vue-3-schedule-recurrence-appointment-in-dst-timezone-forked-2tfdpf?file=/src/App.vue. 

As I mentioned in my previous comment my problem occurs when you will have the first task of the recurrence on the day of the DST. Here is the sample I created based on the most recent link you attached
https://codesandbox.io/s/ej2-vue-3-schedule-recurrence-appointment-in-dst-timezone-forked-gwn6rv?file=/src/App.vue

And here is the image I can see on my screen generated for this sample:
![Screenshot from 2023-07-20 16-12-24](https://github.com/syncfusion/ej2-vue-ui-components/assets/79991096/b2587e0b-7b28-448c-b56a-8e8354e29b26)

The only change I have made is I changed the `startDate` of the repetition to be on the 29th not on the 23rd
